gateway/gateway-requests/src/registration/handshake/shared_key.rs

```diff
@@ -102,6 +103,36 @@ impl SharedKeys {
             .collect()
     }
 
+    pub fn decrypt_tagged(&self, mut enc_data: Vec<u8>) -> Result<Vec<u8>, GatewayRequestsError> {
```

Comment:
I don't think it should be taking ownership (and mutating) incoming data, I'd say `pub fn decrypt_tagged(&self, enc_data: &[u8]) -> Result<Vec<u8>, GatewayRequestsError>` type signature would be more appropriate

Reply:
Done